boring/src/ssl/mod.rs

```diff
@@ -598,6 +598,12 @@ impl SslVersion {
     pub const TLS1_3: SslVersion = SslVersion(ffi::TLS1_3_VERSION as _);
 }
 
+impl From<u16> for SslVersion {
```

Comment:
Should it be `TryFrom` that checks that version is valid?

Reply:
We do the same `impl From<u16>` for `ExtensionType` and `SslSignatureAlgorithm`, should I update those two as well? Although that would be a breaking change

Reply:
I swapped to `TryFrom`, and I can change the other instances of `From` if we're okay with the breaking change

Reply:
Ideally - yes. Let's probably not remove them just yet, but deprecate via an attribute

Reply:
It turns out `From` and `TryFrom` can't coexist: https://github.com/rust-lang/rust/issues/98062

Maybe we should just proceed as is and leave a warning in the very next release notes about this deprecation. Then in the next major bump we can fully swap to `TryFrom`